pkg/metrics/api/api.go

```diff
@@ -0,0 +1,313 @@
+	// ClusterInstallationModificationBootstrapJobName is the variable that stores the metric name for the modification of bootstrap manifests details.
+	ClusterInstallationModificationBootstrapJobName = "cluster_installation_modification_bootstrap_ignition"
+	// ClusterInstallationModificationMasterJobName is the variable that stores the metric name for the modification of master manifests details.
+	ClusterInstallationModificationMasterJobName = "cluster_installation_modification_master_ignition"
+	// ClusterInstallationModificationWorkerJobName is the variable that stores the metric name for the modification of worker manifests details.
+	ClusterInstallationModificationWorkerJobName = "cluster_installation_modification_worker_manifest"
+)
+
+var (
+	// FileCategories is a map of asset names to the categories of the metrics that it belongs to.
+	FileCategories = make(map[string]string)
+	// metricRegistry holds all the MetricBuilder objects for all the metrics that have been designed
+	// and ready for extraction from the installer.
+	metricRegistry = make(map[string]builder.MetricBuilder)
+
+	// CreateCommandMetricName is the map of the targets to the respective metric names. Installer can
+	// look this up for the correct metric name given a target.
+	CreateCommandMetricName = make(map[string]string)
+
+	// CurrentInvocationContext is used to store the current Invocation metric name used for building context.
+	// Invocation data information is gathered from different parts of the installer and it gets hard for the
+	// installer to keep track of the current command and target that is being run. Hence, the current invocation
+	// metric name is stored here in this variable.
+	CurrentInvocationContext string
+
+	// enableMetrics holds the user preference for sending metrics. If the user chooses not to send, this will hold
+	// the value false.
+	enableMetrics bool
+
+	// prometheusURL is the standard URL to send metrics to. The user can override this information to any URL they choose.
+	prometheusURL = "https://localhost:9092/"
+
+	// pushClient holds the URL, jobname and the http client information to send metrics.
+	pushClient pushclient.PushClient
+)
+
+// init holds all the initial setup information like creating all the metrics that are allowed to
+// be extracted, checking for user input about disabling metrics, URL and for creating the Push Client.
+// Must be run before the metircs are collected.
+func init() {
+	enableMetrics = true
+
+	if _, ok := os.LookupEnv("OPENSHIFT_INSTALL_DISABLE_METRICS"); ok {
+		enableMetrics = false
+	}
+
+	if value, ok := os.LookupEnv("OPENSHIFT_INSTALL_METRICS_ENDPOINT"); ok {
+		prometheusURL = value
+	}
+	initializeModificationCategories()
+	initializeCreateCommandMetricName()
+
+	pushClient = pushclient.PushClient{url: prometheusURL, Client: &http.Client{}, JobName: "openshift_installer_metrics"}
```

Comment:
I think would should try and remove any global initialization.

I think we can use the main function to start / initialize any of these